module/module_test.go

```diff
@@ -493,192 +463,124 @@ func TestModularResourceWithoutReconfigure(t *testing.T) {
-	var depNames []string
-	for k := range mock.deps {
-		depNames = append(depNames, motor.Named(k.Name).String())
+	mockConf := &v1.ComponentConfig{
+		Name:  "mymock1",
+		Api:   shell.Subtype.String(),
+		Model: model.String(),
 	}
-	test.That(t, depNames, test.ShouldResemble, deps)
-	test.That(t, mock.config.Motors, test.ShouldResemble,
-		[]string{"rdk:component:motor/motor1"})
-
-	reconfigureDeps := []string{motor.Named("motor2").String()}
-	_, err = m.ReconfigureResource(ctx, &pb.ReconfigureResourceRequest{
-		Config: newSumConf, Dependencies: reconfigureDeps,
-	})
-	test.That(t, err, test.ShouldBeNil)
-
-	// Assert that mock is configured correctly after reconfiguring.
-	var reDepNames []string
-	for k := range mock.deps {
-		reDepNames = append(reDepNames, motor.Named(k.Name).String())
+	mockReconfigConf := &v1.ComponentConfig{
+		Name:  "mymock2",
+		Api:   shell.Subtype.String(),
+		Model: modelWithReconfigure.String(),
 	}
-	test.That(t, reDepNames, test.ShouldResemble, reconfigureDeps)
-	test.That(t, mock.config.Motors, test.ShouldResemble,
-		[]string{"rdk:component:motor/motor2"})
 
-	err = conn.Close()
-	test.That(t, err, test.ShouldBeNil)
-
-	m.Close(ctx)
+	//nolint:dupl
+	t.Run("non-reconfigurable creation", func(t *testing.T) {
+		mockAttrs, err := protoutils.StructToStructPb(MockConfig{
+			Motors: []string{motor.Named("motor1").String()},
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	err = myRobot.Close(ctx)
-	test.That(t, err, test.ShouldBeNil)
-}
+		mockConf.Attributes = mockAttrs
 
-// TestModularResourceWithReconfigure tests that modularly-handled resources
-// with a Reconfigure method are correctly set up when initially added and
-// later reconfigured.
-func TestModularResourceWithReconfigure(t *testing.T) {
-	ctx := context.Background()
-	logger := golog.NewTestLogger(t)
+		validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
+			Config: mockConf,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	cfg := &config.Config{Components: []config.Component{
-		{
-			Name:  "motor1",
-			API:   resource.NewSubtype("rdk", "component", "motor"),
-			Model: resource.NewDefaultModel("fake"),
-		},
-		{
-			Name:  "motor2",
-			API:   resource.NewSubtype("rdk", "component", "motor"),
-			Model: resource.NewDefaultModel("fake"),
-		},
-	}}
+		deps := validateResp.Dependencies
+		test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor1"})
 
-	myRobot, err := robotimpl.RobotFromConfig(ctx, cfg, logger)
-	test.That(t, err, test.ShouldBeNil)
+		_, err = m.AddResource(ctx, &pb.AddResourceRequest{
+			Config: mockConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	parentAddr, err := myRobot.ModuleAddress()
-	test.That(t, err, test.ShouldBeNil)
+		test.That(t, createDeps1, test.ShouldResemble, deps)
+		test.That(t, createConf1.Attributes, test.ShouldResemble, mockConf.Attributes)
+		test.That(t, createConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
```

Comment:
Shouldn't this test the actual values within `ConvertedAttributes`? Not just that it implements `MockConfig{}`? Zeroed out `MockConfig{}` would pass this test but not be valid behavior.